### PR TITLE
Refactor send_input to handle single line comments and lingering output

### LINKED ISSUE #7999: Comment in the last line of Power Shell code block hangs execution
- Azure Data Studio Version: 1.13.0-insider

Steps to Reproduce:

1.Create a notebook with Power Shell kernel
2. Add code block: 
```PowerShell
write "hello"
# last line being a comment hangs the execution
```

### LINKED ISSUE #8277: Comment in Powershell Notebook Code Cell breaks code execution
- Azure Data Studio Version:

Steps to Reproduce:

1. Open a new notebook
2. Change the kernel to Powershell
3. Enter code "Get-DscResource  # put in some comments" into first line of code cell
4. Execute
The Execute button will change to a square and no result will be returned. Any other code in that notebook will not execute until the notebook is closed and reopened. The same code without the comment will execute as expected after restart.

### PR DESCRIPTION
**Pull request by corivera**
Fixes https://github.com/microsoft/azuredatastudio/issues/8277
Fixes https://github.com/microsoft/azuredatastudio/issues/7999

There are a couple of things going on in this change. Originally, I intended to fix the above issues with end of line comments by adding a newline to the end of the input text. That solved the comment problem, but then I started seeing issues with cell execution always returning the last result run. Looks like we continually append text to self.output (even after a command "completes"), which messes up the string parsing to get cell outputs later. My solution here was to clear self.output before running a command. 

Since we're modifying self.output in a multithreaded environment, I also decided to combine the input and output functions and add some locking for good measure.

Any ideas for potential edge case tests here would be appreciated.

### CODE REVIEW

File: powershell_kernel/powershell_proxy.py [hunk 1]
```diff
@@ -25,44 +25,46 @@ def run(self):
...
-
-        self.expected_output_prefix = input.replace('\n', '\n>> ') + '\n'
-        self.expected_output_len = len(self.expected_output_prefix)
-        self.output_prefix_stripped = False
-
-        self._repl.write(input + '\n')
+    def run_command(self, input):
+        self.runCmdLock.acquire()
+        try:
+            self.output = ''
```
Comment (chlafreniere) [inline powershell_kernel/powershell_proxy.py:48]: what if an exception is thrown in the try block?

Comment (corivera) [reply powershell_kernel/powershell_proxy.py]: We have a try-catch block in the kernel.py method that calls this. We have to do some special message wrapping for exceptions there.

Comment (corivera) [reply powershell_kernel/powershell_proxy.py:48]: There's a try-catch block in the kernel do_execute method that calls this one. We have to do some special message wrapping for exceptions there.

**Review by chlafreniere — APPROVED** [review]

File: powershell_kernel/powershell_proxy.py [hunk 2]
```diff
@@ -25,44 +25,46 @@ def run(self):
...
+            self.stop_flag = False
+
+            # Append newline to the original input to handle single line comments on the last line
+            #
+            # Also, for multiline statements we should send 1 extra new line at the end
+            # https://stackoverflow.com/questions/13229066/how-to-end-a-multi-line-command-in-powershell
+            input = '. {\n' + input + '\n}\n'
```
Comment (ellbosch) [inline powershell_kernel/powershell_proxy.py:55]: I noticed a few strings in this method aren't created with string formatting. It's totally optional but I find it helps make strings a little more readable.

File: powershell_kernel/powershell_proxy.py (hunk 1 above)
Comment (ellbosch) [inline powershell_kernel/powershell_proxy.py:48]: I don't think this is a big deal but there's folks out there who think it's best practice to only set class attributes inside `__init__`. It's something that pylint likes to shout about. Interesting discussion [here](https://stackoverflow.com/questions/19284857/instance-attribute-attribute-name-defined-outside-init).

**Review by ellbosch — APPROVED** [review]

File: powershell_kernel/powershell_proxy.py (hunk 1 above)
Comment (corivera) [reply powershell_kernel/powershell_proxy.py:48]: This variable gets modified by a separate "view loop" thread, so it's easier to use a class field here.

File: powershell_kernel/powershell_proxy.py (hunk 2 above)
Comment (corivera) [reply powershell_kernel/powershell_proxy.py:55]: In this case there are some curly braces in the original string, so using format() makes the string even more confusing. I'll leave this one as is.
